Implemented Sanity CMS & More Pages

NickBurnett opened the pull request:
- added join, events, programs, sponsors, and scholarship pages
- added Sanity CMS for all content injections (few hardcoded sections, will change before prod)
- lost the code formatting battle to a formidable opponent...

NickBurnett committed:
Sprint Over

- implemented sanity CMS

- implemented basic dynamic calendar events, although the system sucks. not my system, the exec calendar system >:(

- implemented project overlays to display more information at higher accessibility

- attempted to fix the timeline, keeping that in the backlog for now

- modified components to account for Sanity data architecture

- implemented events and join pages

- no pages have SVG graphics yet but that is not priority at the moment

NickBurnett committed:
More Pages :)

- added sponsor, scholarship, TIP, Projects Program pages

- integrated a program template in Sanity for content injection

- integrated page templates in Sanity for sponsor and scholarship pages

- note that the other team and program pages will be done by the 22nd, as I had no freaking clue the board meeting was moved to Tuesday XD

NickBurnett committed:
Added PrettierRC

WillieCubed committed:
Remove duplicate lint-staged call in pre-commit hook

WillieCubed committed:
Update Prettier config to use semicolons and run Prettier

Nicolas did not like the semicolons. We agreed to disagree. Well,
we agreed that he disagreed with my preference for semicolons.

WillieCubed commented on components/DevProject.js:
```diff
@@ -4,15 +4,101 @@ import styles from '../styles/component/DevProject.module.css';
+      <div className={styles.overlay}>
+        <div className={styles.overlay_card}>
+          <div className={styles.overlay_exit}>
+            <svg
```
Instead of directly embedding SVG into the page, consider making a component.

NickBurnett replied:
This will happen later tonight once I have a good system for SVGs. Resolving so that this can merge faster.

WillieCubed commented on components/ProgramPage.js:
```diff
@@ -0,0 +1,108 @@
+import Info from '../components/Info';
+import ImageCarousel from './ImageCarousel';
+import styles from '../styles/component/ProgramPage.module.css';
+export class Testimonial extends Component {
```
What does this component do?

WillieCubed commented on components/ProgramPage.js:
```diff
@@ -0,0 +1,108 @@
+  }
+}
+
+export default class ProgramPage extends Component {
```
What does this component do? Describe the props.

WillieCubed commented on components/TeamPage.js:
```diff
@@ -5,37 +5,74 @@ import Timeline from './Timeline';
 import styles from '../styles/component/TeamPage.module.css';
 
 export class Profile extends Component {
```
What does this component do? Describe the props.

NickBurnett committed:
Updated README.md to include Sanity documentation

- im using long commit names now willie!

WillieCubed commented:
#### 📝Changed routes:


| [**/events**](https://acmutd-website-kl01a3mow-acmutd.vercel.app/events) | [**/**](https://acmutd-website-kl01a3mow-acmutd.vercel.app/) |
|:-:|:-:|
|<a href="https://acmutd-website-kl01a3mow-acmutd.vercel.app/events"><img src="https://images.weserv.nl?url=https://sjc.microlink.io/ElQPirenKPv4qeGvee1QUFaeWNK1qLbVoJ5WxmLfSU9LzfVykB5LyzP3y3CewrltsSKD8yHS2TMlis1H3ARSjA.png&w=600" alt="Screenshot of /events" width="300"></a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/DohQqBbVPjingeptPMNkTU4l30w2PZTDygkrPbeod26HFUPn1F0Sx6oRHvVsaZtFSXl0J3P3NUOKNaCDuN0lNw.png">(view full size)</a> |<a href="https://acmutd-website-kl01a3mow-acmutd.vercel.app/"><img src="https://images.weserv.nl?url=https://sjc.microlink.io/Thje2dd8_GGSHX2_aduCyz_YpWPdxgc6OlGlnnXzi39v--4TjTGAsuC441fIj95XfW-QvzBpDrM5xvUUaGnG7w.png&w=600" alt="Screenshot of /" width="300"></a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/zTam3a5pCN2XiDBEuM-zMg_oVqWy1wlTxBgj9NvnjOzaqZXPx0LwREOlZ_gRUfuHv81zsAVTrLJm8zBmoM3YTA.png">(view full size)</a> |



| [**/join**](https://acmutd-website-kl01a3mow-acmutd.vercel.app/join) | [**/program/projects**](https://acmutd-website-kl01a3mow-acmutd.vercel.app/program/projects) |
|:-:|:-:|
|<a href="https://acmutd-website-kl01a3mow-acmutd.vercel.app/join"><img src="https://images.weserv.nl?url=https://sjc.microlink.io/VMHxL8dXUDbEz5Apr7oq6zftPKR2R-4v9ifbh6yIa9bbh4FRgcDj3SAi4s4yQWti0XJZ7-kEXTE4Xj_vYwZFgA.png&w=600" alt="Screenshot of /join" width="300"></a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/Hl10Vforlafu-wOqXJyauR-HdWXqiGELjHe_7t-cKDgbMkuSEQ0DWdmnMJDaI8BbeqdPa5AlD5LKzSr4KLz8cA.png">(view full size)</a> |<a href="https://acmutd-website-kl01a3mow-acmutd.vercel.app/program/projects"><img src="https://images.weserv.nl?url=https://sjc.microlink.io/z9ATW-VGjzSeHXJVEYIJp_I8lxrguGmmuXxF0Y8G1_UAgKPRoiLNmXKUttcRvyO6ttkdC7VP6pMti5bdaGpgXg.png&w=600" alt="Screenshot of /program/projects" width="300"></a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/FW3yoYjLE5Y1feRCyJJrFe9UQxtWYIZmH6P_HSgQ88UNJ-ruzehMWD0vFTlFMrVNFNhAKfvF7vgb9xtrII85lA.png">(view full size)</a> |



| [**/program/tip**](https://acmutd-website-kl01a3mow-acmutd.vercel.app/program/tip) | [**/scholarship**](https://acmutd-website-kl01a3mow-acmutd.vercel.app/scholarship) |
|:-:|:-:|
|<a href="https://acmutd-website-kl01a3mow-acmutd.vercel.app/program/tip"><img src="https://images.weserv.nl?url=https://sjc.microlink.io/DWE-AjZfhGbo4Pv_ItLv0Glp5WPnw0PWPC-eVQJpWkYVZyYWYINz83Rsd3PGrrQW7KhX2SyCBemPcXa9VcT3qA.png&w=600" alt="Screenshot of /program/tip" width="300"></a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/VSAP852sgU-T2Om57CmiShWtONKgQgiJBjFsvmyAmHG16az5vhnsj-dLymcnmXXIJmaQJOKIT0rzDaEsYn6Luw.png">(view full size)</a> |<a href="https://acmutd-website-kl01a3mow-acmutd.vercel.app/scholarship"><img src="https://images.weserv.nl?url=https://sjc.microlink.io/Adna8ercB-LCv5A4Eg0Seas3RSdpBJ20Hq4_TdYSyQTuFcQEJZngVTMu-9lO7YTDl_9orJvE2GSXUB4qAkuW9w.png&w=600" alt="Screenshot of /scholarship" width="300"></a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/mY6Ir2drnHfNUqT4Cm4fMcrMPKRBUo8HeLQiN_Xb5UfkdHo40Jjs5RWLml1tchodnV62rsEgkqFDVQml5Vo9Fg.png">(view full size)</a> |



And 5 other routes:
- [**/sponsors**](https://acmutd-website-kl01a3mow-acmutd.vercel.app/sponsors)
- [**/team/dev**](https://acmutd-website-kl01a3mow-acmutd.vercel.app/team/dev)
- [**/team/education**](https://acmutd-website-kl01a3mow-acmutd.vercel.app/team/education)
- [**/team/industry**](https://acmutd-website-kl01a3mow-acmutd.vercel.app/team/industry)
- [**/team/media**](https://acmutd-website-kl01a3mow-acmutd.vercel.app/team/media)

1 deleted route:
- **/dev**

Commit 9071e8883e4108e830cd12fb950132d4509d9e30 (https://acmutd-website-kl01a3mow-acmutd.vercel.app).

NickBurnett commented:
> #### 📝Changed routes:
> [**/events**](https://acmutd-website-2oui55a3b-acmutd.vercel.app/events)	[**/**](https://acmutd-website-2oui55a3b-acmutd.vercel.app/)
> [<img alt="Screenshot of /events" width="300" src="https://camo.githubusercontent.com/e8b65a274a1a2a3f663dbe50e1a0b63b94caa77fed9681655c1841b5d9c344fd/68747470733a2f2f696d616765732e7765736572762e6e6c3f75726c3d68747470733a2f2f736a632e6d6963726f6c696e6b2e696f2f4b7a776c74724e4253715f367a6a30686f5958436347394b7565486a4e686c504462526c53514a347a44764a397750755a506a5a717a43626b326c4734444a766b32576a4233464167324a416234736a4363653539512e706e6726773d363030">](https://acmutd-website-2oui55a3b-acmutd.vercel.app/events)
> [(view full size)](https://images.weserv.nl?url=https://sjc.microlink.io/yit7SGUTEP-amzW7uSNHukJ0dE9T8Fo9c3_eL6dOjRpGjRgAYcybaQiR6p2jAk-pQxTFGn_9TngPLXZoeRiyAg.png)	[<img alt="Screenshot of /" width="300" src="https://camo.githubusercontent.com/48c5dd18772a1a9ae685b24f43d3b4f61b3776848d9b214d39f542c65310a00b/68747470733a2f2f696d616765732e7765736572762e6e6c3f75726c3d68747470733a2f2f736a632e6d6963726f6c696e6b2e696f2f59476d682d735f7673594b5f4a587052534576674a4d76706e4532466e587a4f5f2d754f3175566f37466874344965586759426f33494c525a495549716d6f62536c455742456d4437464d53436544623866694459512e706e6726773d363030">](https://acmutd-website-2oui55a3b-acmutd.vercel.app/)
> [(view full size)](https://images.weserv.nl?url=https://sjc.microlink.io/f1_JLj_gnMY10yttWk1Yu1IxL0ro7sjK_qTLKkyCpncIU9TadxCcbJkK4Fpy6jmA7JvGVYOcLI8NVvOxLgJUvw.png)
> [**/join**](https://acmutd-website-2oui55a3b-acmutd.vercel.app/join)	[**/program/projects**](https://acmutd-website-2oui55a3b-acmutd.vercel.app/program/projects)
> [<img alt="Screenshot of /join" width="300" src="https://camo.githubusercontent.com/d55d29768d3de5deabef7344713e2ad959b73dd55813ae030159b58dd434ac39/68747470733a2f2f696d616765732e7765736572762e6e6c3f75726c3d68747470733a2f2f736a632e6d6963726f6c696e6b2e696f2f35646a33397a623576775a6532686d71414a78556b7968325568537463636f314f6f7931417a3857416c753951376172707231727554335445615a6952504d74722d6e314a32564a4753586c487069435057674a62512e706e6726773d363030">](https://acmutd-website-2oui55a3b-acmutd.vercel.app/join)
> [(view full size)](https://images.weserv.nl?url=https://sjc.microlink.io/pvqCGdq-roF54TISVvQVMd9BfswjllHjKsRr9jyG8_9Mg0jt_S8HSj5myq6rD65ryjav6YUIF7rrZDNkxJmLjg.png)	[<img alt="Screenshot of /program/projects" width="300" src="https://camo.githubusercontent.com/37b535e5e808adb0be3c107f5c0f7694c3940536e6179c348ae12c0b66fe7331/68747470733a2f2f696d616765732e7765736572762e6e6c3f75726c3d68747470733a2f2f736a632e6d6963726f6c696e6b2e696f2f6c7032674c394f313174694c6e47456a434c41736b4d694e63597435304c4f516a41556e6e76512d625671744557675475636e44623661724b666f35485776595433706a4d6965325f494b6a727775455a74785374772e706e6726773d363030">](https://acmutd-website-2oui55a3b-acmutd.vercel.app/program/projects)
> [(view full size)](https://images.weserv.nl?url=https://sjc.microlink.io/fCDdRvgMTI-_KHL_HBUB-m_KQmN62TFCxs9p0irTEbRl6XAH_XeWQHpQLCoKiwbv71_8A46rHJFXrkWa_ZCpkQ.png)
> [**/program/tip**](https://acmutd-website-2oui55a3b-acmutd.vercel.app/program/tip)	[**/scholarship**](https://acmutd-website-2oui55a3b-acmutd.vercel.app/scholarship)
> [<img alt="Screenshot of /program/tip" width="300" src="https://camo.githubusercontent.com/9d17f5a338273f6fe88fe214110a7a7916ba6b1e3b4a9bb1d35711fb582ea720/68747470733a2f2f696d616765732e7765736572762e6e6c3f75726c3d68747470733a2f2f736a632e6d6963726f6c696e6b2e696f2f374a59586d6d592d586c6e326a6b6b6d7130487a33716143344743536f6e4f4b355a6e686a63774c6e774e514b79614d4d6b6b563147455869364871755f35785a72704b6d6a726b5f487050547861666e396f6751772e706e6726773d363030">](https://acmutd-website-2oui55a3b-acmutd.vercel.app/program/tip)
> [(view full size)](https://images.weserv.nl?url=https://sjc.microlink.io/uqPjdwFkvZ-xGqroRi9QFG4rVmSX-UdaaHc1KqMf5LE5kRSlaRJ1togXuYdsidezZxgBT1zdTTNDL2SSKKUtjA.png)	[<img alt="Screenshot of /scholarship" width="300" src="https://camo.githubusercontent.com/e78f721972ab263be6a1dd9b8dda844ce5075f7f857ed16ee24932bc3c0af7e3/68747470733a2f2f696d616765732e7765736572762e6e6c3f75726c3d68747470733a2f2f736a632e6d6963726f6c696e6b2e696f2f3657314134364957516a4f4f574e7659307754796669753570614e744556745f4f4a596e676d5874443854766a30306250634447367455787a3965466f56416d4572564e6b4d5465306b30494c583938423539444b672e706e6726773d363030">](https://acmutd-website-2oui55a3b-acmutd.vercel.app/scholarship)
> [(view full size)](https://images.weserv.nl?url=https://sjc.microlink.io/5jpebq48IWKY8higfoq70ELtpu4OAXD4_KPABF4UXsivNNRhNvnime_GbvH6zVE-iClhBg8YWKNOkjF5IJrFJg.png)
> And 5 other routes:
> 
> * [**/sponsors**](https://acmutd-website-2oui55a3b-acmutd.vercel.app/sponsors)
> * [**/team/dev**](https://acmutd-website-2oui55a3b-acmutd.vercel.app/team/dev)
> * [**/team/education**](https://acmutd-website-2oui55a3b-acmutd.vercel.app/team/education)
> * [**/team/industry**](https://acmutd-website-2oui55a3b-acmutd.vercel.app/team/industry)
> * [**/team/media**](https://acmutd-website-2oui55a3b-acmutd.vercel.app/team/media)
> 
> 1 deleted route:
> 
> * **/dev**
> 
> Commit [75b4731](https://github.com/acmutd/acmutd.github.io/commit/75b4731efcc47d581db1eb446210d16105e1054d) (https://acmutd-website-2oui55a3b-acmutd.vercel.app).

What does this do?

WillieCubed committed:
Add Sanity CLI to devDependencies

This change allows the sanity/ subproject to run the Sanity scripts without
installing the CLI globally.

WillieCubed commented on .prettierrc.js:
```diff
@@ -3,6 +3,6 @@ module.exports = {
   semi: true,
   singleQuote: true,
   tabWidth: 2,
-  trailingComma: 'all',
```
Trailing commas are actually better for version control purposes, so I'd revert this change.

NickBurnett committed:
Addressed basic PR comments for a quick merge

- documentation will happen once this is merged so that development can accelerate

NickBurnett committed:
Added Board SVG
